compiler/qsc_parse/src/expr.rs

```diff
@@ -166,7 +166,18 @@ fn expr_base(s: &mut ParserContext) -> Result<Box<Expr>> {
     } else if token(s, TokenKind::Keyword(Keyword::Fail)).is_ok() {
         Ok(Box::new(ExprKind::Fail(expr(s)?)))
     } else if token(s, TokenKind::Keyword(Keyword::For)).is_ok() {
-        let vars = pat(s)?;
+        let peeked = s.peek();
+        let vars = match pat(s) {
+            Ok(o) => o,
+            Err(e) => {
+                if let (TokenKind::Open(Delim::Paren), ErrorKind::Token(..)) = (peeked.kind, &e.0) {
```

Comment:
Would it be possible to check the error contents for the "in" keyword to make this even more specific?

Reply:
Yes, that should work. I'm thinking about if that would miss any cases...but I think that should work? I'll add some tests to exercise it.

Reply:
Updated in [db2c617](https://github.com/microsoft/qsharp/pull/1828/commits/db2c617546d3843c3390c321a18f8d19a03ed9a4).